api/src/client/dynamics/housing/DynamicsHousingPropertyInterestClient.ts

```diff
@@ -0,0 +1,61 @@
+import { HousingPropertyInterest, HousingPropertyInterestClient } from "@client/dynamics/housing/types";
+import { LogWriterType } from "@libs/logWriter";
+import axios, { AxiosError } from "axios";
+
+export const DynamicsHousingPropertyInterestClient = (
+  logWriter: LogWriterType,
+  orgUrl: string
+): HousingPropertyInterestClient => {
+  const getPropertyInterestsByAddress = async (
+    accessToken: string,
+    address: string
+  ): Promise<HousingPropertyInterest[]> => {
+    const logId = logWriter.GetId();
+    logWriter.LogInfo(`Dynamics Housing Property Interest Client - Id:${logId}`);
+
+    return axios
+      .get(
+        `${orgUrl}/api/data/v9.2/ultra_propertyinterests?$select=createdon,ultra_isfiresafetyproperty,ultra_isbhiregisteredproperty,ultra_streetaddress,ultra_zipcode,ultra_bhinextinspectiondue_date,ultra_bhinextreinspectiondue_state,statecode&$filter=(ultra_streetaddress eq ${address})&$top=10`,
+        {
+          headers: {
+            Authorization: `Bearer ${accessToken}`,
+          },
+        }
+      )
+      .then((response) => {
+        logWriter.LogInfo(
+          `Dynamics Housing Property Interest Client - Id:${logId} - Response: ${JSON.stringify(
+            response.data
+          )}`
+        );
+        return response.data.value.map((response: DynamicsHousingPropertyInterestResponse) => {
+          return {
```

Comment:
[pebble] thoughts on pulling the processing of the response into it's own function below just to reduce clutter within this function

Reply:
That makes sense as it might evolve to be more complicated, will do